File: CS-Examples/02_Data/ImportDataFromArrayList.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace ImportDataFromArrayList
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Create an empty worksheet
            workbook.CreateEmptySheets(1);

            //Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            //Create an ArrayList object
            ArrayList list = new ArrayList();

            //Add strings in list
            list.Add("Spire.Doc for .NET");
            list.Add("Spire.XLS for .NET");
            list.Add("Spire.PDF for .NET");
            list.Add("Spire.Presentation for .NET");

            //Insert array list in worksheet 
            sheet.InsertArrayList(list, 1, 1, true);

            // Specify the name for the resulting Excel file
            string result = "ImportDataFromArrayList_out.xlsx";

            // Save the workbook to the specified file in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(result);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}

File: CS-Examples/02_Data/ImportDataFromArrayList.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace ImportDataFromArrayList
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Create an empty worksheet within the workbook
            workbook.CreateEmptySheets(1)

            ' Get the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Create an ArrayList object
            Dim list As New ArrayList()

            ' Add strings to the ArrayList
            list.Add("Spire.Doc for .NET")
            list.Add("Spire.XLS for .NET")
            list.Add("Spire.PDF for .NET")
            list.Add("Spire.Presentation for .NET")

            ' Insert the ArrayList data into the worksheet starting at cell position (1, 1)
            sheet.InsertArrayList(list, 1, 1, True)

            ' Save the workbook to an Excel file named "ImportDataFromArrayList_out.xlsx" in Excel 2013 format
            Dim result As String = "ImportDataFromArrayList_out.xlsx"
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(result)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
